v2/mapping_services.proto

```diff
@@ -0,0 +1,140 @@
+  string forwarder_id = 2;
+  // LoRa Alliance NetID of the Home Network Member.
+  uint32 home_network_net_id = 3;
+  // New Gateway Visibility for the Home Network Member.
+  GatewayVisibility visibility = 4;
+}
+
+message GetGatewayVisibilityResponse {
+  GatewayVisibility visibility = 1;
+}
+
+service GatewayVisibilityManager {
+  // Get the Default Gateway Visibility.
+  rpc GetDefaultVisibility(GetDefaultGatewayVisibilityRequest) returns (GetGatewayVisibilityResponse);
+  // Set the Default Gateway Visibility.
+  rpc SetDefaultVisibility(SetDefaultGatewayVisibilityRequest) returns (google.protobuf.Empty);
+  // Get the Gateway Visibility with the Home Network.
+  rpc GetHomeNetworkVisibility(GetHomeNetworkGatewayVisibilityRequest) returns (GetGatewayVisibilityResponse);
+  // Set the Gateway Visibility with the Home Network.
+  rpc SetHomeNetworkVisibility(SetHomeNetworkGatewayVisibilityRequest) returns (google.protobuf.Empty);
+}
+
+message UpdateForwarderGatewayRequest {
+  // LoRa Alliance NetID of the Forwarder Member.
+  uint32 forwarder_net_id = 1;
+  // Forwarder identifier issued by the Forwarder Member.
+  string forwarder_id = 2;
+  // Gateway to update.
+  Gateway gateway = 3;
+}
+
+service ForwarderMapper {
+  // Update a gateway.
+  rpc UpdateGateway(UpdateForwarderGatewayRequest) returns (google.protobuf.Empty);
+}
+
+message ListHomeNetworkGatewaysRequest {
+  // LoRa Alliance NetID of the Home Network Member.
+  uint32 home_network_net_id = 1;
+  // Number of items to skip for pagination.
+  uint32 offset = 2;
+  // Limit number items for pagination.
+  // If 0, use the server's default.
+  uint32 limit = 3;
+}
+
+message GetHomeNetworkGatewayRequest {
+  // LoRa Alliance NetID of the Home Network Member.
+  uint32 home_network_net_id = 1;
+  // LoRa Alliance NetID of the Forwarder Member.
+  uint32 forwarder_net_id = 2;
+  // Forwarder identifier issued by the Forwarder Member.
+  string forwarder_id = 3;
+  // Unique identifier of the gateway.
+  uint64 eui = 4;
+}
+
+message SubscribeHomeNetworkGatewaysRequest {
+  // LoRa Alliance NetID of the Home Network Member.
+  uint32 home_network_net_id = 1;
+
+  message ForwarderEUIs {
+    // LoRa Alliance NetID of the Forwarder Member.
+    uint32 forwarder_net_id = 1;
+    // Forwarder identifier issued by the Forwarder Member.
+    string forwarder_id = 2;
+    // Set of unique identifiers of gateways.
+    // Leave empty to subscribe to all gateways of the Forwarder Member.
+    repeated uint64 euis = 3;
+  }
+  // Set of gateways to subscribe to.
+  // Leave empty to subscribe to all gateways.
+  repeated ForwarderEUIs gateways = 2;
+
+  // Shared subscription group (optional).
+  // Queue durability depends on the Mapper configuration.
+  string group = 3;
+}
+
+service HomeNetworkMapper {
+  // List gateways.
+  // This rpc sets the total number of gateways in the x-total-count header.
+  rpc ListGateways(ListHomeNetworkGatewaysRequest) returns (Gateways);
```

Comment:
Let's use a `ListHomeNetworkGatewaysResponse` message here, that includes the `total` field, similar to `ListTenantsResponse`.